src/backend_pgsql.c

```diff
@@ -270,7 +270,7 @@ backend_authenticate(const char *service, const char *user, const char *passwd,
 						}
 					} else {
 						tmp = password_encrypt(options, user, passwd, stored_pw);
-						if (!strcmp(stored_pw, tmp)) {
+						if (tmp != NULL && !strcmp(stored_pw, tmp)) {
 							rc = PAM_SUCCESS;
 						}
 						free (tmp);
```

Comment:
You're still trying to free tmp

Reply:
Right, but that shouldn't matter: free() is explicitly allowed to be called with NULL as a parameter. From the manpage: "If ptr is NULL, no operation is performed."
Would you prefer if I changed to code this way?
```
                                                tmp = password_encrypt(options, user, passwd, stored_pw);
                                                if (tmp != NULL ) {
                                                        if (!strcmp(stored_pw, tmp)) {
                                                                rc = PAM_SUCCESS;
                                                        }
                                                        free (tmp);
                                                }
```